lib/pbench/test/unit/server/database/test_audit_db.py

```diff
@@ -0,0 +1,209 @@
+        attr2 = {"message": "OK", "updated": {"dataset.access": "public"}}
+        ds1 = Dataset(id=1, name="testy", resource_id="md5.1")
+        ds2 = Dataset(id=2, name="tasty", resource_id="md5.2")
+        user1 = User(id=1, username="imme")
+        user2 = User(id=2, username="imyou")
+        with freeze_time("2022-01-01 00:00:00 UTC") as f:
+            root1 = Audit.create(
+                dataset=ds1,
+                user=user2,
+                operation=OperationCode.CREATE,
+                status=AuditStatus.BEGIN,
+            )
+
+            f.tick(delta=datetime.timedelta(seconds=20))
+            root2 = Audit.create(
+                dataset=ds2,
+                user=user1,
+                operation=OperationCode.UPDATE,
+                status=AuditStatus.BEGIN,
+            )
+
+            f.tick(delta=datetime.timedelta(hours=5))
+            Audit.create(root=root1, status=AuditStatus.FAILURE, attributes=attr1)
+
+            f.tick(delta=datetime.timedelta(minutes=40))
+            Audit.create(root=root2, status=AuditStatus.SUCCESS, attributes=attr2)
+
+        # Try some queries. The mocked query returns the ordered list of
+        # filters passed to the database engine: we want to confirm that the
+        # expected filters were run, and there's no point in trying to mock the
+        # behavior of those filters.
+
+        Audit.query(user_name="imme")
+        self.session.check_session(filters=["user_name=imme"])
+
+        Audit.query(object_name="testy")
+        self.session.check_session(filters=["object_name=testy"])
+
+        Audit.query(dataset=ds1)
+        self.session.check_session(
+            filters=["audit.object_type = AuditType.DATASET, audit.object_id = md5.1"]
+        )
+
+        Audit.query(timestamp=dateutil.parser.parse("2022-01-01 00:00:00 UTC"))
+        self.session.check_session(filters=["timestamp=2022-01-01 00:00:00+00:00"])
+
+        Audit.query(start=dateutil.parser.parse("2022-01-01 04:00:00 UTC"))
+        self.session.check_session(
+            filters=["audit.timestamp >= 2022-01-01 04:00:00+00:00"]
+        )
+
+        Audit.query(
+            start=dateutil.parser.parse("2022-01-01 01:00:00 UTC"),
+            end=dateutil.parser.parse("2022-01-01 04:00:00 UTC"),
+        )
+        self.session.check_session(
+            filters=[
+                "audit.timestamp >= 2022-01-01 01:00:00+00:00",
+                "audit.timestamp <= 2022-01-01 04:00:00+00:00",
+            ]
+        )
+
+        Audit.query(
+            name="test",
+            start=dateutil.parser.parse("2022-01-01 01:00:00 UTC"),
+            end=dateutil.parser.parse("2022-01-01 04:00:00 UTC"),
+        )
+        self.session.check_session(
+            filters=[
+                "name=test",
+                "audit.timestamp >= 2022-01-01 01:00:00+00:00",
+                "audit.timestamp <= 2022-01-01 04:00:00+00:00",
+            ]
+        )
```

Comment:
At the point when I reviewed this, `user` and `operation` were (positional) parameters to `Audit.query()`, but there weren't any unit tests which supplied them as arguments in the call.  This seemed like a gap, but you have since removed them (although, the docstring still describes `operation`), so that issue is mostly resolved (modulo the docstring...).

The unit tests were also testing parameters `name` and `user_name`, but, since the function handles all non-positional parameters the same, testing them _both_ seemed redundant.  It also seemed slightly suspicious since, at the time, we had `user`, `name`, **and** `user_name` which seemed like maybe there was some overlap/redundancy/staleness (but...perhaps the first of those is a record or ID, the second refers to a dataset and not a username, and so the third doesn't overlap with either of the preceding...but I've lost track).

I see that `start`, `end`, and `dataset` require special handling, so it makes sense to separate them from the other `kwargs`.  As you say, listing them as positional arguments doesn't prevent the caller from specifying them using keywords.  However, I think we can get the best of both worlds:
```
    @staticmethod
    def query(
        *,
        start: Optional[datetime] = None,
        end: Optional[datetime] = None,
        dataset: Optional[Dataset] = None,
        **kwargs,
    ) -> "list[Audit]":	    ) -> "list[Audit]":
```
Adding the `*` forces all the parameters to be keyword-only, but `start`, `end`, and `dataset`, while optional, will not be included in `kwargs`.